pkg/kubectl/resource/visitor.go

```diff
@@ -564,6 +564,9 @@ func RetrieveLatest(info *Info, err error) error {
 	if err != nil {
 		return err
 	}
+	if runtime.IsListType(info.Object) {
+		return fmt.Errorf("watch is only supported on a single resource, but a list of resources is found")
```

Comment:
This error is misleading given:
https://github.com/kubernetes/kubernetes/issues/15665#issuecomment-148216664

how about:
on individual resources and resource collections